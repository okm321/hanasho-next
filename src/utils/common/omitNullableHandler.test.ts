import { expectType } from 'tsd'
import { omitNullableHandler } from './omitNullableHandler'

describe('omitNullableHandler', () => {
  const items = ['A', 'B', 'C', null, undefined] as const
  it(`omitNullable to ${items.map((item) => `${item}`)}`, () => {
    const omittedNullableItems = items.filter(omitNullableHandler)

    expectType<Array<'A' | 'B' | 'C'>>(omittedNullableItems)
    expect(omittedNullableItems).toMatchSnapshot()
  })
})
