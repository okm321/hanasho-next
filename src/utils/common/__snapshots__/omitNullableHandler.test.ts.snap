// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omitNullableHandler omitNullable to A,B,C,null,undefined 1`] = `
[
  "A",
  "B",
  "C",
]
`;
